### PR TITLE
user profile

### PR DESCRIPTION
**Pull request by NguyenQuynh1994**

### CODE REVIEW

File: resources/views/user/show.blade.php [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    <div class="row">
+        <div class="col-md-10 col-md-offset-1">
+            <div class="panel panel-info">
+                <div class="panel-heading">{{ trans('label.my_profile') }}</div>
+
+                <div class="panel-body">
+                    @include('common.errors')
+                    {!! Form::open([
+                        'class' => 'form-horizontal'
+                    ]) !!}
```
Comment (NgTheThao) [inline resources/views/user/show.blade.php]: :cry: cái này thì cần gì phải xuống dòng ghê vậy, form ko có action gì à? mà show blade sao lại có form?

Comment (NguyenQuynh1994) [inline resources/views/user/show.blade.php]: cái này là show nên ko có action ạ, mà bên dưới e có dùng mấy cái Form:text, label nên dùng Form:open với Form:colse, Thực ra thì ko dùng cũng được ạ

File: resources/views/user/edit.blade.php [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    <div class="row">
+        <div class="col-md-10 col-md-offset-1">
+            <div class="panel panel-info">
+                <div class="panel-heading">{{ trans('label.my_profile') }}</div>
+
+                <div class="panel-body">
+                    @include('common.errors')
+                    {!! Form::open([
+                        'method' => 'PUT',
+                        'url' => 'user/profiles/' . $user->id,
```
Comment (NgTheThao) [inline resources/views/user/edit.blade.php]: nên dùng action nhé

File: resources/views/layouts/app.blade.php [hunk 3]
```diff
@@ -58,6 +74,6 @@
     <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/2.2.3/jquery.min.js" integrity="sha384-I6F5OKECLVtK/BL+8iSLDEHowSAfUo76ZL9+kGAgTRdiByINKJaqTPH/QVNS1VDb" crossorigin="anonymous"></script>
     <script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.6/js/bootstrap.min.js" integrity="sha384-0mSbJDEHialfmuBBQP6A4Qrprq5OVfW37PRR3j5ELqxss1yVqOtnepnHVP9aJ7xS" crossorigin="anonymous"></script>
     <script src="http://ajax.aspnetcdn.com/ajax/jquery.validate/1.15.0/jquery.validate.min.js"></script>
-    <script src="js/validate.js"></script>
+    <script src="{{ asset('js/validate.js') }}"></script>
```
Comment (NgTheThao) [inline resources/views/layouts/app.blade.php]: sửa cả đoạn trên nữa `{!! Html::style() !!}`

File: resources/views/layouts/app.blade.php [hunk 4]
```diff
@@ -10,7 +10,7 @@
     <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Lato:100,300,400,700">
     <!-- Styles -->
     <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.6/css/bootstrap.min.css" integrity="sha384-1q8mTJOASx8j1Au+a5WDVnPi2lkFfwwEAa8hDDdjZlpLegxhjVME1fgjWPGmkzs7" crossorigin="anonymous">
-    <link rel="stylesheet" type="text/css" href="css/app.css">
+    <link href="{{ asset('/css/app.css') }}" rel="stylesheet">
```
Comment (NgTheThao) [inline resources/views/layouts/app.blade.php]: sửa cả đoạn trên nữa `{!! Html::style() !!}`

File: resources/views/user/show.blade.php (hunk 1 above)
Comment (NguyenQuynh1994) [inline resources/views/user/show.blade.php]: thực ra form ko có action và cũng ko cần dùng đến form ạ, e sẽ xóa đi ạ

File: app/Http/Requests/UpdateUserRequest.php [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+     * Get the validation rules that apply to the request.
+     *
+     * @return array
+     */
+    public function rules()
+    {
+        $rule = config('common.limit');
+
+        return [
+            'avatar' => "image|max:{$rule['image_size']}",
```
Comment (NgTheThao) [inline app/Http/Requests/UpdateUserRequest.php]: ở js em có validate `extension : "jpeg|png|bmp|gif|svg",` thì ở đây validate nó nt nữa

File: resources/views/user/edit_password.blade.php [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+                        <div class="form-group">
+                            {!! Form::label('password', trans('label.password'), ['class' => 'col-md-3 control-label']) !!}
+                            <div class="col-md-6">
+                                {!! Form::password('password', ['class' => 'form-control', 'id' => 'password']) !!}
+                            </div>
+                        </div>
+                        <div class="form-group">
+                            {!! Form::label('newPassword', trans('label.new_password'), ['class' => 'col-md-3 control-label']) !!}
+                            <div class="col-md-6">
+                                {!! Form::password('newPassword', ['class' => 'form-control', 'id' => 'newPassword']) !!}
```
Comment (NgTheThao) [inline resources/views/user/edit_password.blade.php]: id thif nên đặt là `new-password`, ở dưới là `new-password-confirmation`, class ở form cũng đặt như-thế-này-nhé
mà chỗ này là change pwd khi đã đăng nhập rồi à>? ko phải chức năng reset pwd à? :D

File: app/Http/Controllers/User/UserController.php [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+    public function update(UpdateUserRequest $request, $id)
+    {
+        $validator = Validator::make($request->all(), $request->rules());
+
+        if ($validator->fails()) {
+            return redirect()->route('user.profiles.show', $id)->withErrors($validator);
+        }
+
+        try {
+            $this->userRepository->updateInfo($request);
```
Comment (NgTheThao) [inline app/Http/Controllers/User/UserController.php]: có cần thiết phải try catch không?

File: resources/views/user/edit_password.blade.php (hunk 6 above)
Comment (NguyenQuynh1994) [inline resources/views/user/edit_password.blade.php]: reset là quên password còn đây là user muốn change password

File: app/Http/Controllers/User/UserController.php (hunk 7 above)
Comment (NguyenQuynh1994) [inline app/Http/Controllers/User/UserController.php]: ở userRepository e có throw ra exception khi mà không update được nên bên này e cho vào trong try/cactch để bắt exception

Comment (NguyenQuynh1994) [inline app/Http/Controllers/User/UserController.php]: lúc đầu bên UserRepository e có throw ra exception nên bên Controller e để try catch

File: app/Http/Controllers/User/UserController.php [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+            return redirect()->route('user.profiles.show', $id)->withErrors($validator);
+        }
+
+        $user = $this->userRepository->updateInfo($request);
+
+        if (isset($user['error'])) {
+            return redirect()->back()->withErrors($user['error']);
+        }
+
+        return redirect()->route('user.profiles.show', $id)->withErrors(trans('message.update_user_success'));
```
Comment (hungcoj) [inline app/Http/Controllers/User/UserController.php]: Update success mà đưa ra withErrors à :dash:

File: app/Http/Controllers/User/UserController.php [hunk 9]
```diff
@@ -0,0 +1,144 @@
...
+
+        $user = $this->userRepository->changePassword($request);
+
+        if (isset($user['error'])) {
+            return redirect()->back()->withErrors($user['error']);
+        }
+
+        Auth::login($user);
+
+        return redirect()->route('user.profiles.show', $id)->withErrors(trans('message.change_password_success'));
```
Comment (hungcoj) [inline app/Http/Controllers/User/UserController.php]: Đây nữa :|

**Comment by hungcoj** [comment]
Ok!

**Comment by NgTheThao** [comment]
ok
